test/common/router/config_impl_test.cc

```diff
@@ -849,13 +852,134 @@ TEST_F(RouterMatcherHashPolicyTest, HashHeaders) {
   {
     Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
     Router::RouteConstSharedPtr route = config.route(headers, 0);
-    EXPECT_FALSE(route->routeEntry()->hashPolicy()->generateHash("", headers).valid());
+    EXPECT_FALSE(
+        route->routeEntry()->hashPolicy()->generateHash("", headers, add_cookie_nop_).valid());
   }
   {
     Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
     headers.addCopy("foo_header", "bar");
     Router::RouteConstSharedPtr route = config.route(headers, 0);
-    EXPECT_TRUE(route->routeEntry()->hashPolicy()->generateHash("", headers).valid());
+    EXPECT_TRUE(
+        route->routeEntry()->hashPolicy()->generateHash("", headers, add_cookie_nop_).valid());
+  }
+  {
+    Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/bar", "GET");
+    Router::RouteConstSharedPtr route = config.route(headers, 0);
+    EXPECT_EQ(nullptr, route->routeEntry()->hashPolicy());
+  }
+}
+
+TEST_F(RouterMatcherHashPolicyTest, HashCookieNoTtl) {
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<Upstream::MockClusterManager> cm;
+  route_config_.mutable_virtual_hosts(0)
+      ->mutable_routes(0)
+      ->mutable_route()
+      ->add_hash_policy()
+      ->mutable_cookie()
+      ->set_name("hash");
+  ConfigImpl config(route_config_, runtime, cm, true);
+
+  EXPECT_FALSE(config.usesRuntime());
+
+  {
```

Comment:
Not in the style guide (so optional) but Harvey and I encourage test comments so folks less familiar with the code can easily see what's going on.  Like
// With no cookie set no hash will be generated
// With non-configured cookies set, the hash still will not be generated
// Finally with the configured "hash" cookie set, a hash will be generated.